client/src/wayland_obj/layer_shell.rs

```diff
@@ -0,0 +1,168 @@
+use std::{cell::RefCell, rc::Rc};
+
+use wayland_client::{protocol::wl_buffer::WlBuffer, GlobalImplementor, NewProxy, Proxy};
+
+pub(super) use wayland_protocols::wlr::unstable::layer_shell::v1::client::{
```

Comment:
i dunno about the rest of these, but this can just be `use`.

`pub(super)` makes whatever is on the right usable by the parent module. Generally you should just be able to use `pub`, the amount of times you will need `pub(super)` seems fairly rare.